src/backend/distributed/executor/multi_utility.c

```diff
@@ -290,6 +291,19 @@ multi_ProcessUtility(Node *parsetree,
 		}
 	}
 
+	/* handle DDL on the worker node */
+	if (!schemaNode)
```

Comment:
I was worried about that too and I agree with your opinion, since this is what we do in the current mx.